libs/traits/src/liquidity_pools.rs

```diff
@@ -12,14 +12,17 @@
 // GNU General Public License for more details.
 
 use frame_support::{dispatch::DispatchResult, weights::Weight};
-use sp_runtime::DispatchError;
+use sp_runtime::{app_crypto::sp_core::H160, DispatchError};
 use sp_std::vec::Vec;
 
 pub type Proof = [u8; 32];
 
 /// An encoding & decoding trait for the purpose of meeting the
 /// LiquidityPools General Message Passing Format
-pub trait LPEncoding: Sized {
+pub trait LpMessage: Sized {
+	type Domain;
+	type SerializableDomain;
```

Comment:
@lemunozm Apologies for wrecking this trait. Regard this as WIP to get to an MVP. I am not sure yet how to properly solve this. I tried using generics for the two forwarding functions but didn't manage to please the compiler. If you have a recommendation, please speak up!

Reply:
I am not sure if I follow, but I do not see the `SerializabbleDomain` associated type used by the trait

Reply:
Regarding the domain, why not directly use a `Domain` type?

EDIT: ah because it's in `cfg-types`. I think maybe `Domain` should be treated as primitive, because all pallets use them directly. Still not having a well defined place where to put types lead us to not accurate designs, that's a pity 😢. Hopefully, after this audit we'll have time to do a cool refactor.

By now I think it's ok using `Domain` as associated type, but I think `SerializableDomain` can be removed, right?

Reply:
> I am not sure if I follow, but I do not see the `SerializabbleDomain` associated type used by the trait

Artifact of a previous implementation indeed!

Reply:
Removed: [e5dedb2](https://github.com/centrifuge/centrifuge-chain/pull/1971/commits/e5dedb2f696e470abd94fe3481ab6cb9dd316ca9)